### PR TITLE
UI: Rename input profile

### LINKED ISSUE #6507: [Feature Request]: (Qt) Input profile renaming
### Description

Ability to rename an input profile along with automatic updating of its name on all the games it's being used in.

### Reason

Since a user could potentially use an input profile in more games (such as if as they added more games to their library), they might feel the need to change its name at some point to reflect it for the purpose of their own convenience.

### Examples

-

### PR DESCRIPTION
**Pull request by rtavarez98**
### Description of Changes
Adds feature to change a controller input profile and automatically updates the name in all games using the profile.

### Rationale behind Changes
Resolves #6507

### Suggested Testing Steps
Rename an existing input profile that's assigned to at least one game and check if it's been updated in a game's settings.

### CODE REVIEW

**Comment by Mrlinkwii** [comment]
A very small thing , can the commit have the prefix of UI:

**Comment by Mrlinkwii** [comment]
seems to working good here on linux , its seems the Mac ci has failed

**Comment by TheLastRar** [comment]
Mac CI fails with;
```
../../pcsx2-qt/Settings/ControllerSettingsWindow.cpp:204:52: error: no member named 'recursive_directory_iterator' in namespace 'std::filesystem'
        for (const auto& game_settings : std::filesystem::recursive_directory_iterator(game_settings_dir))
```
So you are missing the `<filesystem>`
Not that we don't use std:;filesystem anywhere else, and I would like additional testing on windows with unicode paths.


To better match existing PCSX2 code, I would suggest you instead opt for using our own FileSystem functions (in `common/FileSystem.h`). We have the following function which I think will suit your needs;
`FindFiles(const char* path, const char* pattern, u32 flags, FindResultsArray* results)`

**Review by Mrlinkwii — APPROVED** [review]

**Review by JordanTheToaster — APPROVED** [review]

File: pcsx2-qt/Settings/ControllerSettingsWindow.ui [hunk 1]
```diff
@@ -113,6 +113,16 @@
          </property>
         </widget>
        </item>
+	   <item>
```
Comment (F0bes) [inline pcsx2-qt/Settings/ControllerSettingsWindow.ui:116]: Indenting with tabs instead of spaces.

**Review by F0bes — CHANGES_REQUESTED** [review]

**Review by F0bes — APPROVED** [review]
Code LGTM

**Pull request merged**